src/relax/ir/block_builder.cc

```diff
@@ -141,6 +130,8 @@ class BlockBuilderImpl : public BlockBuilderNode {
     }
 
     context_mod_->Update(gv, function);
+    // Force update struct info
+    gv->struct_info_ = GetStructInfo(function);
```

Comment:
one potential issue here is that the caller still relies on the assumption of previous function sigature, so automatically updating this may not always be desirable, is there any motivating usecase here?